larpandoracontent/LArHelpers/LArMCParticleHelper.cc

```diff
@@ -216,6 +216,42 @@ const MCParticle *LArMCParticleHelper::GetMainMCParticle(const ParticleFlowObjec
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+const MCParticle *LArMCParticleHelper::GetBestMCParticle(const ParticleFlowObject *const pPfo)
+{
+    CaloHitList caloHitList;
+    LArPfoHelper::GetCaloHits(pPfo, TPC_VIEW_U, caloHitList);
+    LArPfoHelper::GetCaloHits(pPfo, TPC_VIEW_V, caloHitList);
+    LArPfoHelper::GetCaloHits(pPfo, TPC_VIEW_W, caloHitList);
+
+    const MCParticleWeightMap mcParticleWeightMap(GetMCParticleWeightMap(caloHitList));
```

Comment:
Is it better to create an empty MCParticleWeightMap locally, then pass it by non const reference to the function, which fills it? Or some equivalent, using move semantics.